Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml

```diff
@@ -0,0 +1,131 @@
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Logs type Description in Barracuda WAF
+  Link: https://campus.barracuda.com/product/webapplicationfirewall/doc/78810008/types-of-logs/
+Description: |
+  This ASIM parser supports normalizing Barracuda WAF logs to the ASIM Web Session normalized schema.
+ParserName: ASimWebSessionBarracudaWAF
+EquivalentBuiltInParser: _ASim_WebSession_BarracudaWAF
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let SeverityLookup = datatable (severity: int, EventSeverity: string)
+    [
+      0, "High", 
+      1, "High", 
+      2, "High", 
+      3, "Medium",
+      4, "Low",
+      5, "Low", 
+      6, "Informational",
+      7, "Informational" 
+  ];
+  let EventResultWFLookup = datatable (
+      Action_s: string,
+      EventResult_WF: string,
+      DvcAction: string
+  )
+      [
+      "LOG", "Success", "ALLOW",
+      "DENY", "Failure", "DENY",
+      "WARNING", "Success", "WARNING"
+  ];
+  let EventTypeLookup = datatable (LogType_s: string, EventType: string, EventOriginalType: string )
+      [
+      "WF", "HTTPsession", "Web Firewall",
+      "TR", "WebServerSession", "Access"
+  ];
+  let parser=(disabled: bool=false) {
+      barracuda_CL
+      | where not(disabled) and (LogType_s == "WF" or LogType_s == "TR")
+      | lookup EventResultWFLookup on Action_s
+      | lookup EventTypeLookup on LogType_s
+      | extend
+          severity = toint(Severity_s)
+      | lookup SeverityLookup on severity
+      | extend
+          Dst = column_ifexists("ServerIP_s", ""),
+          EventCount = toint(1),
+          EventProduct = "WAF",
+          EventSchema = "WebSession",
+          EventSchemaVersion = "0.2.6",
+          EventVendor = "Barracuda",
+          status_code = toint(column_ifexists("HTTPStatus_s", ""))
+      | extend EventResult_TR = case(
+                                status_code between (200 .. 299),
+                                "Success", 
+                                status_code between (400 .. 599),
+                                "Failure",
+                                status_code between (300 .. 399),
+                                "Partial",
+                                "NA"
+                            )
+      | extend
+          Dvc = column_ifexists("UnitName_s", ""),
+          EventResult = iff(LogType_s == "TR", EventResult_TR, EventResult_WF),
+          EventStartTime = unixtime_milliseconds_todatetime(tolong(DeviceReceiptTime_s)),
+          DstIpAddr = column_ifexists("ServerIP_s", ""),
+          SrcIpAddr = column_ifexists("ClientIP_s", ""),
+          DstBytes = tolong(column_ifexists("BytesReceived_d", "")),
+          DstPortNumber = toint(column_ifexists("ServerPort_d", "")),
+          HttpCookie = column_ifexists("Cookie_s", ""),
+          HttpReferrer = column_ifexists("Referer_s", ""),
+          HttpRequestBodyBytes = tolong(column_ifexists("BytesReceived_d", "")),
```

Comment:
Should this be the opposite? Request is what we sent and Response is what we receive.

Reply:
We mapped based on the documentation of barracuda as below.
https://campus.barracuda.com/product/webapplicationfirewall/doc/92767349/exporting-log-formats/
<img width="726" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/98145046/fb178ecd-cb06-417a-927e-2f4fc81ca2c9">